controllers/cloud.redhat.com/makers/maker.go

```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
 	}}
 }
 
+func (m *Maker) makeJob(job crd.Job, app *crd.ClowdApp, hash string) error {
+
+	nn := types.NamespacedName{
+		Name:      fmt.Sprintf("%v-%v", app.Name, job.Name),
+		Namespace: m.Request.Namespace,
+	}
+
+	c := batch.CronJob{}
+	err := m.Client.Get(m.Ctx, nn, &c)
+
+	update, err := utils.UpdateOrErr(err)
+	if err != nil {
+		return err
+	}
+
+	initJob(m.App, m.Env, &c, nn, job, hash)
+
+	if err := update.Apply(m.Ctx, m.Client, &c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func initJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, jc *batch.CronJob, nn types.NamespacedName, job crd.Job, hash string) {
+	labels := app.GetLabels()
+	labels["pod"] = nn.Name
+	app.SetObjectMeta(jc, crd.Name(nn.Name), crd.Labels(labels))
+
+	pod := job.PodSpec
+
+	jc.Spec.Schedule = job.Schedule
```

Comment:
If we are assuming that the `Schedule` will always be set on the spec, we should not configure the field with `omitempty`.  We can always add this back later when we support the more generic `Job` resource.

Reply:
Yeh, that was why I made it optional to allow the use of specifying Jobs

Reply:
But you are right